rotkehlchen/chain/evm/decoding/curve/curve_cache.py

```diff
@@ -79,41 +61,46 @@ class CurvePoolData(NamedTuple):
     underlying_coins: list[ChecksumEvmAddress] | None
 
 
-def read_curve_pools_and_gauges() -> tuple[dict[ChecksumEvmAddress, list[ChecksumEvmAddress]], set[ChecksumEvmAddress]]:  # noqa: E501
-    """Reads globaldb cache and returns:
-    - A set of all known curve pools addresses.
-    - A set of all known curve gauges addresses.
+def get_read_curve_pools_and_gauges(
+        chain_id: ChainID,  # pylint: disable=unused-argument  # TODO: use it for the cache keys
+) -> Callable[..., tuple[dict[ChecksumEvmAddress, list[ChecksumEvmAddress]], set[ChecksumEvmAddress]]]:  # noqa: E501
+    """Returns a function to reads globaldb cache and returns:
+    - A set of all known curve pools addresses for the given chain.
+    - A set of all known curve gauges addresses for the given chain.
 
-    Doesn't raise anything unless cache entries were inserted incorrectly.
+    Returned functions doesn't raise anything unless cache entries were inserted incorrectly.
     """
-    with GlobalDBHandler().conn.read_ctx() as cursor:
-        curve_pools_lp_tokens = globaldb_get_general_cache_values(
-            cursor=cursor,
-            key_parts=(CacheType.CURVE_LP_TOKENS,),
-        )
-        curve_pools = {}
-        curve_gauges = set()
-        for lp_token_addr in curve_pools_lp_tokens:
-            pool_address = globaldb_get_unique_cache_value(
-                cursor=cursor,
-                key_parts=(CacheType.CURVE_POOL_ADDRESS, lp_token_addr),
-            )
-            if pool_address is None:
-                continue
-            gauge_address_data = globaldb_get_unique_cache_value(
+    def read_curve_pools_and_gauges() -> tuple[dict[ChecksumEvmAddress, list[ChecksumEvmAddress]], set[ChecksumEvmAddress]]:  # noqa: E501
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            curve_pools_lp_tokens = globaldb_get_general_cache_values(
                 cursor=cursor,
-                key_parts=(CacheType.CURVE_GAUGE_ADDRESS, pool_address),
+                key_parts=(CacheType.CURVE_LP_TOKENS,),
```

Comment:
shouldn't this take the chainId? Also I'm not sure how you do it but use the number instead of the string